converter/ffmpeg.py

```diff
@@ -540,15 +540,25 @@ def hwaccel_decoder(self, video_codec, hwaccel):
 
     def encoder_formats(self, encoder):
         prefix = "Supported pixel formats:"
-        formatline = next((line.strip() for line in self._get_stdout([self.ffmpeg_path, '-hide_banner', '-h', 'encoder=%s' % encoder]).split('\n')[1:] if line and line.strip().startswith(prefix)), "")
-        formats = formatline.split(":")
-        return formats[1].strip().split(" ") if formats and len(formats) > 0 else []
+        format_line = next((line.strip() for line in self._get_stdout([self.ffmpeg_path, '-hide_banner', '-h', f"encoder={encoder}"]).split('\n')[1:] if line and line.strip().startswith(prefix)), "")
+
+        if format_line:
+            formats = format_line.split(":")
+            if len(formats) > 1:
+                return formats[1].strip().split(" ")
+        else:
+            return []
 
     def decoder_formats(self, decoder):
         prefix = "Supported pixel formats:"
-        formatline = next((line.strip() for line in self._get_stdout([self.ffmpeg_path, '-hide_banner', '-h', 'decoder=%s' % decoder]).split('\n')[1:] if line and line.strip().startswith(prefix)), "")
-        formats = formatline.split(":")
-        return formats[1].strip().split(" ") if formats and len(formats) > 0 else []
+        format_line = next((line.strip() for line in self._get_stdout([self.ffmpeg_path, '-hide_banner', '-h', f"decoder={decoder}"]).split('\n')[1:] if line and line.strip().startswith(prefix)), "")
```

Comment:
These changes avoid an index out-of-bounds-errors when trying to get formats for invalid codecs.